nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/util/SecureNiFiConfigUtil.java

```diff
@@ -0,0 +1,191 @@
+import java.security.GeneralSecurityException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class SecureNiFiConfigUtil {
+
+    private static final int CERT_DURATION_DAYS = 60;
+    public static final String LOCALHOST_IP = "127.0.0.1";
+    private static final String LOCALHOST_NAME = "localhost";
+
+    private SecureNiFiConfigUtil() {
+
+    }
+
+    /**
+     * If HTTPS is enabled (nifi.web.https.port is set), but the keystore file specified in nifi.security.keystore
+     * does not exist, this will generate a key pair and self-signed certificate, generate the associated keystore
+     * and truststore and write them to disk under the configured filepaths, generate a secure random keystore password
+     * and truststore password, and write these to the nifi.properties file.
+     * @param nifiPropertiesFilename The filename of the nifi.properties file
+     * @param cmdLogger The bootstrap logger
+     * @throws IOException can be thrown when writing keystores to disk
+     * @throws RuntimeException indicates a security exception while generating keystores
+     */
+    public static void configureSecureNiFiProperties(String nifiPropertiesFilename, Logger cmdLogger) throws IOException, RuntimeException {
+                final File propertiesFile = new File(nifiPropertiesFilename);
+        final Properties nifiProperties = loadProperties(propertiesFile);
+
+        if (!nifiProperties.getProperty(NiFiProperties.WEB_HTTPS_PORT, "").isEmpty()) {
+            String keystorePath = nifiProperties.getProperty(NiFiProperties.SECURITY_KEYSTORE, "");
+            String truststorePath = nifiProperties.getProperty(NiFiProperties.SECURITY_TRUSTSTORE, "");
+            if (!keystorePath.isEmpty() && !truststorePath.isEmpty()) {
+                File keystore = new File(keystorePath);
+                File truststore = new File(truststorePath);
+                if (!keystore.exists() && !truststore.exists()) {
+                    TlsConfiguration tlsConfiguration = null;
+                    cmdLogger.info("Generating default self-signed certificates, keystore and truststore for secure Apache NiFi configuration.  " +
+                            "This certificate will expire in {} days, and it is recommended to acquire your own certificates in order to properly secure Apache NiFi.", CERT_DURATION_DAYS);
+                    try {
+                        Pair<String[], String[]> subjectAlternativeNames = getSubjectAlternativeNames(nifiProperties);
+                        tlsConfiguration = KeyStoreUtils.createTlsConfigAndNewKeystoreTruststore(StandardTlsConfiguration
+                                .fromNiFiProperties(nifiProperties), CERT_DURATION_DAYS, subjectAlternativeNames.getLeft(),
+                                subjectAlternativeNames.getRight());
+                    } catch (IOException e) {
```

Comment:
Given that the error messages do not provide different details, recommend combining `IOException` and `GeneralSecurityException` in a multi-catch block:
```suggestion
                    } catch (final IOException|GenerallSecurityException e) {
```